shared/video-sdk/develop/live-streaming-over-multiple-channels/project-implementation/react-native.mdx

```diff
@@ -0,0 +1,240 @@
+
+    ``` ts
+    onChannelMediaRelayEvent: code => {
+        switch (code) {
+            case ChannelMediaRelayEvent.RelayEventNetworkDisconnected: // RELAY_EVENT_NETWORK_DISCONNECTED
+                console.log(
+                'User disconnected from the server due to a poor network connection.',
+                );
+                break;
+            case ChannelMediaRelayEvent.RelayEventNetworkConnected: // RELAY_EVENT_NETWORK_CONNECTED
+                console.log('Network reconnected');
+                break;
+            case ChannelMediaRelayEvent.RelayEventPacketJoinedSrcChannel: // RELAY_EVENT_PACKET_JOINED_SRC_CHANNEL
+                console.log('User joined the source channel');
+        }
+    },
+    ```
+
+### Join multiple channels
+
+The alternate approach to multi-channel live streaming is joining multiple channels. In this section, you learn how to implement joining a second channel in your <Vpl k="CLIENT" />.
+
+#### Implement the user interface
+
+In this example, you need a button to join and leave a second channel. In `<View style={styles.btnContainer}>`, add the following code after `<Text onPress={leave} style={styles.button}>Leave</Text>`:
+``` html
+<Text onPress={multipleChannels} style={styles.button}>
+    Join Second Channel
+</Text>
+```
+
+#### Handle the system logic
+
+In your project, import the relevant libraries and declare the required variables.
+
+1.  **Add the required libraries**
+
+    To implement media relay, import the corresponding <Vg k="COMPANY" /> modules. In `App.tsx`, add the following imports before `createAgoraRtcEngine,`:
+
+    ``` ts
+    ChannelMediaOptions,
+    IRtcEngineEx,
+    RtcConnection,
+    ```
+
+    Remove the library `IRtcEngine` as it will be replaced with `IRtcEngineEx`.
+
+2.  **Declare the variables you need**
+
+    To join and manage a second channel, in `App.tsx`, add the following variables to declarations:
+
+    ``` ts
+    var rtcSecondConnection: RtcConnection;
+    const secondChannelName = 'kishan2';
+    const secondChannelUid = 100; // Uid for the second channel
+    const secondChannelToken =
+    '007eJxTYLj0Tuqzd9iUa1O13Otsj23S7r+h8MPFeJ8d20vH9vcmrV8UGAzNk5PMU5NNUk1TLExMTSwtUlIN0sxNU1MTk02MzI0TGTdIJtczSidvltnAysgAgSA+O0N2ZnFGYp4RAwMAAKghWA==';
+    var isSecondChannelJoined = false; // Track connection state of the second channel
+    ```
+
+#### Implement joining multiple channels
+
+To add multi-channel functionality to your <Vpl k="CLIENT" />, take the following steps:
+
+1.  **Use the multi-channel enabled `IRtcEngineEx` interface**
+
+    In the Get Started code, the <Vg k="ENGINE" /> instance was declared as `IRtcEngine`. To implement multi-channel functionality, you use the multi-channel interface of the <Vg k="ENGINE" /> `IRtcEngineEx`. In order to switch to the multi-user interface, do the following:
+
```

Comment:
Please update the indenting for the substeps so the code is rendered inside the step and the numbering works. 

(The code example needs to have a space after the step and be indented )